File: leetcode3/MinimumOperationstoCollectElements/a.js
// Copyright (C) 2023 by iamslash

// 59ms 48.75% 43.6MB 46.81%
// hash set
// O(N) O(N)
var minOperations = function(nums, k) {
    let n = nums.length;
    let numSet = new Set();
    for (let i = n - 1; i >= 0; --i) {
        if (nums[i] <= k) {
            numSet.add(nums[i]);
        }
        if (numSet.size == k) {
            return n - i;
        }
    }
    return -1;    
};

File: leetcode3/MinimumOperationstoCollectElements/a.py
# -*- coding: utf-8 -*-
# Copyright (C) 2023 by iamslash

from typing import List

# 46ms 51
# hash set
# O(N) O(N)
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        n = len(nums)
        numSet = set()
        for i in range(n - 1, -1, -1):
            if nums[i] <= k:
                numSet.add(nums[i])
            if len(numSet) == k:
                return n - i
        return -1
        
